### PR TITLE
T3561: router-advert: add support for specific routes

### PR DESCRIPTION
**Pull request by mark-royds**
## Change Summary

Add vyos support for radvd `route` statement

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3561

## Component(s) name

service router-advert

## Proposed changes

add support for the `route` directive 
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: data/templates/router-advert/radvd.conf.tmpl [hunk 1]
```diff
@@ -30,6 +30,17 @@ interface {{ iface }} {
     AdvOtherConfigFlag {{ 'on' if interface[iface].other_config_flag is defined else 'off' }};
     AdvRetransTimer {{ interface[iface].retrans_timer }};
     AdvCurHopLimit {{ interface[iface].hop_limit }};
+{%     for route in interface[iface].route %}
```
Comment (c-po) [inline data/templates/router-advert/radvd.conf.tmpl]: Please add an `if` condition to check if the `route` key is set.

File: data/templates/router-advert/radvd.conf.tmpl [hunk 2]
```diff
@@ -30,6 +30,17 @@ interface {{ iface }} {
...
+    route {{ route }} {
+{%     if interface[iface].route[route].valid_lifetime is defined %}
```
Comment (c-po) [inline data/templates/router-advert/radvd.conf.tmpl]: Please indent this by two more whitespaces as this is nested inside the `for` loop.

File: src/conf_mode/service_router-advert.py [hunk 3]
```diff
@@ -86,7 +95,7 @@ def verify(rtradv):
 
                 if not (int(valid_lifetime) > int(preferred_lifetime)):
                     raise ConfigError('Prefix valid-lifetime must be greater then preferred-lifetime')
-
+
```
Comment (c-po) [inline src/conf_mode/service_router-advert.py]: Please drop this whitespace line.

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by c-po** [comment]
LGTM

**Review by c-po — APPROVED** [review]

**Review by dmbaturin — APPROVED** [review]

**Pull request merged**